### DISCUSSION TITLE
Prettier / Formatter support

### DISCUSSION THREAD

**Discussion by jasikpark** [markdoc/markdoc#299] [Q&A] [answered]
Is there formatter support for markdoc? I suppose the regular markdown/mdx support gets you most of the way there.. docusaurus has been using similar syntax for forever 🤷

**Comment by mfix-stripe** [comment, marked as answer]
Hey @jasikpark! Prettier actually works for Markdoc in almost all cases out of the box.

However, we also built a [Markdoc formatter function](https://github.com/markdoc/markdoc/blob/main/src/formatter.ts) as part of the core library. You can just call:
```js
const ast = Markdoc.parse(source)

const prettified = Markdoc.format(ast);
```
to get a prettified version of the original source. 

Hope that helps!

**Reply by jasikpark** [reply]
I guess I would need to build a custom script to make that work as an npm script then if prettier didn't work thx

**Reply by mfix-stripe** [reply]
Try out prettier and let me know how it goes. We are planning on eventually building and releasing a VSCode extension that will handle this formatting in the future.

**Reply by jasikpark** [reply]
If prettier doesn't a custom prettier plugin would be preferred so that I can use `prettier --check .` as a CI formatting step

**Reply by jasikpark** [reply]
ah, https://docusaurus.io/docs/markdown-features/admonitions#usage-with-prettier might be relevant to y'all as well
